### PR TITLE
KZOO-310: Allow devices to have an optional addresses field

### PR DESCRIPTION
**Pull request by harenson**
Tests for https://github.com/2600hz/kazoo-crossbar/pull/392

### CODE REVIEW

File: src/seq/seq_devices.erl [hunk 1]
```diff
@@ -112,6 +114,51 @@ seq_kcro_1114() ->
...
+                 ],
+    EmerAddr = kz_json:from_list_recursive([{<<"emergency">>, AddrFields}]),
+    DeviceJObj = kzd_devices:set_addresses(new_device(), EmerAddr),
+    CreateResp = pqc_cb_devices:create(API, AccountId, DeviceJObj),
+    lager:info("created device ~s", [CreateResp]),
+    CreatedDevice = kz_json:get_json_value(<<"data">>, kz_json:decode(CreateResp)),
+    kz_json:are_equal(kzd_devices:addresses(CreatedDevice), EmerAddr),
+
+    %% When any of the emergency address' fields is missing, the request should fail.
+    {FirstKey, _} = hd(AddrFields),
```
Comment (jamesaimonetti) [inline src/seq/seq_devices.erl]: `[{FirstKey, _} | AddrFields2] = AddrFields,` should work too, right?

File: src/seq/seq_devices.erl [hunk 2]
```diff
@@ -112,6 +114,51 @@ seq_kcro_1114() ->
...
+    AddrFields2 = tl(AddrFields), %% Copy without first element.
+    EmerAddr2 = kz_json:from_list_recursive([{<<"emergency">>, AddrFields2}]),
+    DeviceJObj2 = kzd_devices:set_addresses(new_device(), EmerAddr2),
+    _FailResp = {'error', ErrorRespEnc} = pqc_cb_devices:create(API, AccountId, DeviceJObj2),
+    lager:info("expected failure response: ~p", [_FailResp]),
+    ExpErr = kz_json:from_list_recursive([{<<"addresses.emergency.", FirstKey/binary>>
+                                          ,[{<<"required">>
+                                            ,[{<<"value">>,<<"region">>}
```
Comment (jamesaimonetti) [inline src/seq/seq_devices.erl]: Isn't this assuming the "region" key will always be first in the list? So really this is just testing "region" is required. What if you took the keys from the list comprehension above into a var, then did a fold over them, `IncompleteAddrFields = kz_json:delete_key(Key, AddrFields)` and tried to save, then checked the error resp for the Key that was deleted?

File: src/seq/seq_devices.erl (hunk 1 above)
Comment (harenson) [reply src/seq/seq_devices.erl]: Good idea.

File: src/seq/seq_devices.erl (hunk 2 above)
Comment (harenson) [reply src/seq/seq_devices.erl]: Good catch, the test was not failing just because nothing was matching `kz_json:are_equal/2`'s result 🤦🏾

**Pull request merged**